runtime/assembly/modules/storage.ts

```diff
@@ -36,4 +38,48 @@ export namespace Storage {
         const value64: u64 = Serialiser.serialiseResultwOutRetain(value);
         ext_storage_set_version_1(key64, value64);
     }
+    /**
+     * Gets the given key from storage, placing the value into a buffer and returning the number of bytes that the
+     * entry in storage has beyond the offset. 
+     * ArrayBuffer has fixed length of 20.
+     * Returns Option instance with populated ArrayBuffer concatenated to the result from storage_read
+     * @param key key of the data to get
+     * @param offset offset beyond which the value is copied
+     */
+    export function read(key: u8[], offset: i32): Option<u8[]> {
+        const key64: u64 = Serialiser.serialiseResultwOutRetain(key);
+        const valueOut = new ArrayBuffer(20);
+        let valuePtr = changetype<usize>(valueOut);
+        let valueSize = valueOut.byteLength;
+        const valueOut64 = ((valueSize as u64) << 32) | valuePtr;
+
+        const result = ext_storage_read_version_1(key64, valueOut64, offset);
+        
+        const ptrSize: i32[] = Serialiser.separatePointerSize(result);
+        let valueu8: u8[] = Serialiser.deserialiseInput(ptrSize[0], ptrSize[1]);
+
+        let buff = Uint8Array.wrap(valueOut);
+        let res = Utils.toU8Array(buff);
+        
+        return Option.isArraySomething(valueu8) ? new Option<u8[]>(valueu8.slice(1).concat(res)): new Option<u8[]>(null);
+    }
+    /**
+     * Clears the storage of the given key and its value.
+     * @param key key of the data to clear
+     */
+    export function clear(key: u8[]): void {
+        const key64: u64 = Serialiser.serialiseResultwOutRetain(key);
+        ext_storage_clear_version_1(key64);
+    }
+
+    /**
+     * Checks whether the given key exists in storage.
+     * @param key key of the data to check
+     */
+    export function exists(key: u8[]): u8[] {
```

Comment:
I think that it would be better to return `bool` value `true/false` that states whether the key exists or not. 
That way it will be more dev friendly and usable.
This means returning simply the result as bool.